src/pthread/thread.cpp

```diff
@@ -556,6 +560,18 @@ namespace boost
 #elif defined(BOOST_HAS_UNISTD_H) && defined(_SC_NPROCESSORS_ONLN)
         int const count=sysconf(_SC_NPROCESSORS_ONLN);
         return (count>0)?count:0;
+#elif defined(__VXWORKS__)
+        cpuset_t set =  ::vxCpuEnabledGet();
+  #ifdef __DCC__
+        int i;
+        for( i = 0; set; ++i)
+        {
+           set &= set -1;
+        }
+        return(i);
+  #else  
+        return (__builtin_popcount(set) );
```

Comment:
Does it means that when we use VxWorks, we have always a compiler that has this intrinsic?
With with compiler has you tested?

Reply:
VxWorks is packaged with one of 4 validated compilers depending on the board support package and processor, so I know that Intel ICC, GCC, and llvm/clang have __builtin_popcount() and  Diab compiler (__DCC_) does not.  I didn't find a Boost macro for popcount() but I agree its a cleaner solution.

Reply:
Okay, I accept the PR. We will adapt if someone else need something else.

Reply:
Thanks for the PR.